src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,65 @@
 package mate.academy;
 
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private static final int THREAD_POOL_SIZE = 1;
+    private static final int SHUTDOWN_TIMEOUT_MINUTES = 2;
+    private final ThreadPoolExecutor executor;
+    private final Set<EventListener> listeners;
 
+    public EventManager() {
+        this.executor = new ThreadPoolExecutor(
+                THREAD_POOL_SIZE,
+                THREAD_POOL_SIZE,
+                0L,
+                TimeUnit.MILLISECONDS,
+                new LinkedBlockingQueue<>()
+        );
+        this.listeners = new CopyOnWriteArraySet<>();
     }
 
-    public void deregisterListener(EventListener listener) {
+    public void registerListener(EventListener listener) {
+        if (executor.isShutdown()) {
+            listeners.clear();
+            return;
```

Comment:
not sure it makes sense to clear them here and not in the shutdown method. A simple return here should be enough